go_lib/dependency/cr/cr.go

```diff
@@ -154,8 +153,7 @@ func (r *client) ListTags(ctx context.Context) ([]string, error) {
 
 	if r.options.timeout > 0 {
 		// add default timeout to prevent endless request on a huge amount of tags
-		ctxWTO, cancel := context.WithTimeout(ctx, r.options.timeout)
-		defer cancel()
+		ctxWTO, _ := context.WithTimeout(ctx, r.options.timeout) // nolint:govet
```

Comment:
```suggestion
        // Warning!: don't use cancel() in the defer func here. Otherwise *v1.Image outside this function would be inaccessible due to cancelled context, while reading layers, for example.
		ctxWTO, _ := context.WithTimeout(ctx, r.options.timeout) // nolint:govet
```